client/components/Run/index.jsx

```diff
@@ -33,27 +35,46 @@ class Run extends Component {
     }
 
     async componentDidMount() {
+        const {
+            getRun,
+            linkCohortToRun,
+            location: { search },
+            setCohortUserRole,
+            setRun
+        } = this.props;
         const { cohortId, scenarioId } = this.state;
-        const run = await this.props.getRun(scenarioId);
-
-        if (cohortId && run) {
-            const cohort = await this.props.linkCohortToRun(cohortId, run.id);
-
-            if (cohort) {
-                const { id, users } = cohort;
-                const {
-                    user: { username }
-                } = this.props;
-
-                if (!users.find(user => username === user.username)) {
-                    // For now we'll default all unknown
-                    // users as "participant".
-                    await this.props.setCohortUserRole({
-                        id,
-                        role: 'participant'
-                    });
+        const run = await getRun(scenarioId);
+
+        if (run) {
```

Comment:
Why don't we combine the if-statements here anymore? ie why a separate nested check for cohortId?

Reply:
Because there are two separate conditions within the `if (run) {` consequent body now 🤷🏼‍♂️